### PR TITLE
Chat socket

### PR DESCRIPTION
**Pull request by PassaChr**
adds the posibility to chat with the backend with a socket so there is no need for repeted requests for answers.
this allows bidirectional communication between client and amy

Pull request status:
- [x] Code conventions
- [ ] Doku required and added
- [X] Ready for review

### CODE REVIEW

File: chat-socket/src/main/java/de/unistuttgart/iaas/amyassist/amy/socket/ChatServer.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	
+	/**
+	 * the dialog handler
+	 */
+	@Reference DialogHandler handler;
+
+	/**
+	 * the logger
+	 */
+	@Reference Logger logger;
```
[the remark this answers is not in the document]
Comment (Legion2) [reply chat-socket/src/main/java/de/unistuttgart/iaas/amyassist/amy/socket/ChatServer.java]: refactor the code

File: Dockerfile [hunk 2]
```diff
@@ -26,9 +26,11 @@ COPY --from=source /src/.docker/config /app/config
 
 WORKDIR /app
 
+ENV AMY_SERVER_CONFIG_CHATSERVER_SOCKET_PORT 6660
```
Comment (Legion2) [inline Dockerfile]: this must be `AMY_SOCKET_CONFIG_CHATSERVER_SOCKET_PORT`

**Comment by Bensge** [comment]
@PassaChr this PR does not deal with the WebApp, right? The WebApp won't be able to use this because TCP sockets are not available in browser environments

File: pom.xml [hunk 3]
```diff
@@ -35,6 +35,7 @@
 		<amy.utility.version>0.6.0-Snapshot</amy.utility.version>
 		<amy.registry.version>0.6.0-Snapshot</amy.registry.version>
 		<amy.remote-sr.version>0.6.0-Snapshot</amy.remote-sr.version>
+    <amy.chatsocket.version>0.6.0-Snapshot</amy.chatsocket.version>
```
Comment (Legion2) [inline pom.xml]: format

File: chat-socket/src/main/java/de/unistuttgart/iaas/amyassist/amy/socket/ChatWebSocket.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		this.handler = handler;
+	}
+	
+	/**
+	 * @see org.java_websocket.server.WebSocketServer#onClose(org.java_websocket.WebSocket, int, java.lang.String, boolean)
+	 */
+	@Override
+	public void onClose(WebSocket conn, int code, String reason, boolean remote) {
+		this.logger.debug("{} closed connection", conn);
+		this.logger.debug("Code was {}. Reason was {}.",code,reason	);
```
Comment (Legion2) [inline chat-socket/src/main/java/de/unistuttgart/iaas/amyassist/amy/socket/ChatWebSocket.java]: format

**Comment by Bensge** [comment]
Can we get this merged ASAP? Thanks, I depend on this for my next task

**Comment by Legion2** [comment]
@Bensge you can fix the code convention violations

**Pull request merged**